src/common/src/util/value_encoding/mod.rs

```diff
@@ -15,81 +15,45 @@
 //! Value encoding is an encoding format which converts the data into a binary form (not
 //! memcomparable).
 
-use bytes::Buf;
-use serde::Deserialize;
+use bytes::{Buf, BufMut};
+use chrono::{Datelike, Timelike};
 
 use crate::error::{ErrorCode, Result, RwError};
 use crate::types::{
-    deserialize_datum_from, deserialize_datum_not_null_from, serialize_datum_into,
-    serialize_datum_not_null_into, DataType, Datum, Decimal, ScalarImpl,
+    DataType, Datum, Decimal, IntervalUnit, NaiveDateTimeWrapper, NaiveDateWrapper,
+    NaiveTimeWrapper, OrderedF32, OrderedF64, ScalarImpl, ScalarRefImpl,
 };
 
+pub mod error;
+use error::ValueEncodingError;
+
 /// Serialize datum into cell bytes (Not order guarantee, used in value encoding).
 pub fn serialize_cell(cell: &Datum) -> Result<Vec<u8>> {
-    let mut serializer = value_encoding::Serializer::new(vec![]);
-    if let Some(ScalarImpl::Decimal(decimal)) = cell {
-        return serialize_decimal(decimal);
+    let mut buf: Vec<u8> = vec![];
+    if let Some(datum) = cell {
+        buf.put_u8(1);
+        serialize_value(datum.as_scalar_ref_impl(), &mut buf)
+    } else {
+        buf.put_u8(0);
```

Comment:
I'm thinking if we can simply make null elements empty? We can return vec![]. You may have a try, but I don't know whether there would be bugs in our system that disallows this.